lxd/container_lxc.go

```diff
@@ -6377,7 +6696,7 @@ func (c *containerLXC) fillNetworkDevice(name string, m types.Device) (types.Dev
 	}
 
 	// Fill in the MAC address
-	if m["nictype"] != "physical" && m["hwaddr"] == "" {
+	if m["nictype"] != "physical" && m["hwaddr"] == "" && m["type"] != "infiniband" {
```

Comment:
So I'm not sure I understand this check. If we don't want to generate and change the MAC for an SR-IOV infiniband device, why are we fine with doing it for an ethernet one? Seems inconsistent.

Reply:
Hm? This check reads like it prevents changing the Mac for both nictypes Infiniband devices.

Last I tested you could change the Mac of a generic sriov device but Infiniband sriov didn't let you do it. I'm not sure if they are using the same kernel codepaths.

Reply:
Ah, ok, interesting.